docs/aws/audit/appsyncmonitoring/rules/appsync_cache_encryption_at_rest_remediation.mdx

```diff
@@ -2,6 +2,128 @@
+   - Choose the AWS Key Management Service (KMS) key you want to use for encryption. You can select an existing key or create a new one.
+   - Save the changes.
+
+By following these steps, you ensure that the data stored in the AppSync cache is encrypted at rest, enhancing the security of your application.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that encryption at rest is enabled for AWS AppSync cache using the AWS CLI, follow these steps:
+
+1. **Check Current AppSync API Configuration:**
+   First, you need to identify the AppSync API for which you want to enable encryption at rest. Use the following command to list all AppSync APIs and get their IDs.
+
+   ```sh
+   aws appsync list-graphql-apis
+   ```
+
+2. **Describe the AppSync API:**
+   Once you have the API ID, describe the specific AppSync API to check its current configuration.
+
+   ```sh
+   aws appsync get-graphql-api --api-id <your-appsync-api-id>
+   ```
+
+3. **Update AppSync API to Enable Encryption:**
+   If encryption at rest is not enabled, you can update the AppSync API to enable it. Use the following command to update the API configuration. Replace `<your-appsync-api-id>` with your actual API ID and `<kms-key-id>` with your KMS key ID.
+
+   ```sh
+   aws appsync update-graphql-api --api-id <your-appsync-api-id> --name <your-api-name> --authentication-type API_KEY --log-config '{"fieldLogLevel": "ALL", "cloudWatchLogsRoleArn": "<your-cloudwatch-logs-role-arn>"}' --user-pool-config '{"userPoolId": "<your-user-pool-id>", "awsRegion": "<your-region>", "defaultAction": "ALLOW"}' --xray-enabled --encryption-at-rest '{"kmsKeyId": "<kms-key-id>"}'
+   ```
+
+4. **Verify the Update:**
+   Finally, verify that the encryption at rest has been enabled by describing the AppSync API again.
+
+   ```sh
+   aws appsync get-graphql-api --api-id <your-appsync-api-id>
+   ```
+
+By following these steps, you can ensure that encryption at rest is enabled for your AWS AppSync cache using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that encryption at rest is enabled for AWS AppSync cache using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by directly configuring the `~/.aws/credentials` file.
+
+3. **Create a Python Script**:
+   Write a Python script to enable encryption at rest for your AppSync cache. Below is an example script:
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon AppSync
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>